examples/gallery/symbols/patterns.py

```diff
@@ -0,0 +1,76 @@
+r"""
+Bit and Hachure Patterns
+------------------------
+
+PyGMT allows using bit or hachure patterns via the ``fill`` parameter
+or similar parameters:
+
+- Symbols and polygons: :meth:`pygmt.Figure.plot` via ``fill``
+- Histogram bars or sectors: :meth:`pygmt.Figure.histogram` or
+  :meth:`pygmt.Figure.rose` via ``fill``
+- Anomalies: :meth:`pygmt.Figure.wiggle` via ``fillpositive``
+  and ``fillnegative``
+- Land and water masses: :meth:`pygmt.Figure.coast` via ``land``
+  and ``water``
+- Focal mechanisms: :meth:`pygmt.Figure.meca` via ``G`` and ``E``
```

Comment:
We need to add aliases meca's `-G` and `-E` options.

Reply:
Should I do this first in a separate PR?
Are there any suggestions or preferences for the aliases? Maybe `fillcompression` for **-G** and `fill_extension` for **-E** (similar to `fillpositive` and `fillpositive` of `pygmt.Figure.wiggle`)? However, they are inconsistent, as one would be with hyphen and one without. So, maybe the other way around is better, i.e., `compressionfill` and `extensionfill` (similar to `uncertaintyfill` of `pygmt.Figure.velo`).

Reply:
Would prefer the second option to have consistent namings.

Reply:
I started working on this in PR #2345.

Reply:
I also updated the list at the beginning of this example with the aliases for **E** and **G** in commit https://github.com/GenericMappingTools/pygmt/pull/2329/commits/25c37771728cd0c41aa04d76fcb20fd2b0b35726.